Src/Support/GoogleApis.Auth.DotNet4/OAuth2/GoogleCredential.cs

```diff
@@ -87,6 +87,28 @@ public static Task<GoogleCredential> GetApplicationDefaultAsync()
         }
 
         /// <summary>
+        /// Returns the Application Default Credentials (ambient credentials) for the application, synchronously.
+        /// </summary>
+        /// <remarks>
+        /// See <see cref="GetApplicationDefaultAsync"/> for more details on the Application Default Credentials. If you
+        /// are writing an application using asynchronous operations, it is encouraged that you use that method. However,
+        /// if you are writing an application using synchronous operations this method is safe to call, even from a thread
+        /// associated with a single-thread synchronization context such as a Windows Forms UI. The call will block until
+        /// the credentials are available (or an exception is thrown) but will not attempt re-entrancy within the
+        /// synchronization context. So for example, if you call this from a Windows Forms UI thread, the UI will freeze
+        /// until the method completes, but it will not deadlock.
+        /// </remarks>
+        /// <returns>The Application Default Credentials (ambient credentials) for the application.</returns>
+        public static GoogleCredential GetApplicationDefault()
+        {
+            // Everywhere we await a task internally, we use ConfigureAwait(false). Therefore even if this is called on
```

Comment:
I agree it should be possible to add some form of test to verify this doesn't deadlock, but it's a thin ice. If it's not complicating things too much, I'd prefer the Task.Run approach, that seems to be easier to get right. Also, such implementation clearly signals that the method that uses it is kind of a hack and should be used carefully.